### PR TITLE
Changes to UserPreference Logic

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Improve Logic for managing UserPreference
- Add automated UI tests.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Account/TestAuthHandler.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                new Claim(ClaimConstants.Avatar, "https://avatars.githubusercontent.com/u/53356347?v=4"),
+                new Claim(ClaimConstants.Name, "Azure SDK Bot"),
+                new Claim(ClaimConstants.Email,"azuresdkengsysteam@microsoft.com"),
+                new Claim(ClaimConstants.Orgs, "Azure"),
+            };
+
+            var identity = new ClaimsIdentity(claims, "TestUser");
+            var principal = new ClaimsPrincipal(identity);
+            var ticket = new AuthenticationTicket(principal, "TestUser");
+            var result = AuthenticateResult.Success(ticket);
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Account/TestAuthHandler.cs:32]: Should we add additional check to ensure this is only succeed when running as localhost?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs [hunk 2]
```diff
@@ -35,46 +36,37 @@ public IndexPageModel(ReviewManager manager, UserPreferenceCache preferenceCache
             int CurrentPage, int? PreviousPage, int? NextPage) PagedResults { get; set; }
 
         public async Task OnGetAsync(
-            List<string> search = null, List<string> languages=null, List<string> state =null,
-            List<string> status =null, List<string> type =null, int pageNo=1, int pageSize=_defaultPageSize, string sortField=_defaultSortField)
+            IEnumerable<string> search = null, IEnumerable<string> languages=null, IEnumerable<string> state =null,
+            IEnumerable<string> status =null, IEnumerable<string> type = null, int pageNo=1, int pageSize=_defaultPageSize, string sortField=_defaultSortField)
         {
-            if (search.Count == 0 && languages.Count == 0 && state.Count == 0 && status.Count == 0 && type.Count == 0)
+            if (!search.Any() && !languages.Any() && !state.Any() && !status.Any() && !type.Any())
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs:42]: won't this fail is any of these args has default null value?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs [hunk 3]
```diff
@@ -35,46 +36,37 @@ public IndexPageModel(ReviewManager manager, UserPreferenceCache preferenceCache
             int CurrentPage, int? PreviousPage, int? NextPage) PagedResults { get; set; }
 
         public async Task OnGetAsync(
-            List<string> search = null, List<string> languages=null, List<string> state =null,
-            List<string> status =null, List<string> type =null, int pageNo=1, int pageSize=_defaultPageSize, string sortField=_defaultSortField)
+            IEnumerable<string> search = null, IEnumerable<string> languages=null, IEnumerable<string> state =null,
+            IEnumerable<string> status =null, IEnumerable<string> type = null, int pageNo=1, int pageSize=_defaultPageSize, string sortField=_defaultSortField)
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs:40]: Do we really need null as deafault?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs (hunk 2 above)
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Index.cshtml.cs:42]: https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.any?view=net-6.0

File: src/dotnet/APIView/APIViewWeb/Startup.cs [hunk 4]
```diff
@@ -99,7 +102,14 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<LanguageService, SwiftLanguageService>();
             services.AddSingleton<LanguageService, XmlLanguageService>();
 
-            services.AddAuthentication(options =>
+            if (Environment.IsDevelopment())
```
Comment (praveenkuttappan) [inline src/dotnet/APIView/APIViewWeb/Startup.cs:105]: I have another comment about this above. My only worry is  we are leaving a backdoor if we happen to enable dev mode in any of the deployed test env in the future. So adding an extra check will avoid that. One another option is to use key based authentication which can be passed from internal env and in local testing.

**Review by praveenkuttappan — COMMENTED** [review]
Please see comments above

File: .azure-pipelines/apiview.yml [hunk 5]
```diff
@@ -168,6 +167,44 @@ stages:
...
+                  else
+                  {
+                    Write-Host "APIView not yet started"
+                  }
+                  $timeout -= 2
+                }
+                while(($timeout -eq 0) -or ($apiViewHost.Count -eq 0))
+              displayName: 'Start APIView Application'
+              env:
+                APIVIEW_BLOB__CONNECTIONSTRING: $(apiview-blob-connectionstring)
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply .azure-pipelines/apiview.yml:191]: I will address this is a subsequent [issue](https://github.com/Azure/azure-sdk-tools/issues/4196) right now there should be nothing problematic the way I have written the test (unless all reviews are somehow deleted).

File: src/dotnet/APIView/APIViewWeb/Models/UserPreferenceModel.cs [hunk 6]
```diff
@@ -2,26 +2,66 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using CsvHelper.Configuration.Attributes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Newtonsoft.Json;
```
[the remark this answers is not in the document]
Comment (chidozieononiwu) [reply src/dotnet/APIView/APIViewWeb/Models/UserPreferenceModel.cs:6]: Not sure what you mean but Newtonsoft.lJson is used in almost all the models

**Review by praveenkuttappan — APPROVED** [review]
Looks good

**Pull request merged**